src/components/CommentsList.jsx

```diff
@@ -11,12 +11,12 @@ const CommentsList = ({article_id, comments, setComments}) => {
                 setComments(res.comments)
                 setIsLoading(false)
             })
-    }, [])
+    }, [comments])
```

Comment:
Using comments here as a dependency and also setting them in the same useEffect() is not a good idea as currently you have a serious infinite loop. Instead you could use a new state (for example one that indicates that a comment is being deleted with a boolean value).